### ISSUE TITLE
Is there a multiline message support?

### ISSUE THREAD

**Issue by Drugoy** [djeedjay/DebugViewPP#209]
Does DebugViewPP support multiline messages shown as a single message?

That's what I really lacked in DebugView (that I have just switched from), yet seems like DebugViewPP works the same way: it cuts each line into a separate message.

**Comment by janwilmans** [comment]
Yes, just uncheck 'View->Auto newline'

**Comment by Drugoy** [comment]
Uhm, that didn't help: the messages are still chopped into multiple ones.
<img src="https://i.imgur.com/K8wh95I.png">

Also, what's ``<flush>``?

**Comment by Drugoy** [comment]
Also, there's a ~5sec lag between 'message 1 line 1' and the next message, although they are sent at the same time (yes, I'm sure + tested with outputdebug).

**Comment by janwilmans** [comment]
yes, I understand, I was wrong, I the auto-newline automatically accepts messages that do not have a newline but you want to ignore embedded newlines in a message.

The <flush> message is generated when the process dies and messages not ending in a newline and still in the buffer.

The way it is currently implemented you can construct lines from multiple messages, which kindof the reverse of what you are asking.

**Comment by janwilmans** [comment]
I've put up a link for you:
http://www.myquest.nl/sites/debugview/DebugView++Drugoy_release.zip
this is an experimental version that does not split up message by their embedded newlines, however, it messes up the .dblog file. How would you want to deal with the newlines? convert them to spaces?

**Comment by Drugoy** [comment]
I'm not sure I understand you right, but if I did - I'd like to criticize it :)
Let's discuss things slowly and with examples, please.
> I was wrong, I the auto-newline automatically accepts messages that do not have a newline but you want to ignore embedded newlines in a message.

Sorry, I don't understand. Could you give an example of what you mean?

> The "flush" message is generated when the process dies and messages not ending in a newline and still in the buffer.

Do you mean that normally the debugger expects each message to end with a new line character (CR/LF/CRLF)?
Is this some kind of a standard/common advice or where did you take that from?

> I've put up a link for you:
http://www.myquest.nl/sites/debugview/DebugView++Drugoy_release.zip
this is an experimental version that does not split up message by their embedded newlines, however, it messes up the .dblog file.

Err, again I'm not sure you got me right, or it doesn't work the way it should.
It works like this:
<img src="https://i.imgur.com/XLb4EId.png">
(the first result is with 'auto new line' option enabled, and the second one is with that option disabled).

> How would you want to deal with the newlines? convert them to spaces?

No, I'd better have them shown somehow like this:
<img src="http://download-codeplex.sec.s-msft.com/Download?ProjectName=tracespy&DownloadId=238214">

**Comment by janwilmans** [comment]
Ok, clear, @djeedjay will take a look on how to implement it.

About the newline behaviour that the 'debugger expects ', I'm not sure there are any 'rules' defined for this behaviour.

**Comment by Drugoy** [comment]
Is there a chance this issue would get actually fixed?
I see that the development has stalled, which saddens me a lot, because if not this issue and issue #208 - I'd be so happy to switch from DebugView to DebugViewPP.

**Comment by janwilmans** [comment]
@djeedjay had some thoughts on how to implement it, I myself do not know how to get this behaviour into the UI framework we use, but I will ask if he is interested to create it...
